app/js/components/tabs/tabcomponents/programmeComponent.jsx

```diff
@@ -535,7 +536,7 @@ class ProgrammeComponent extends Component {
               
               <div className="form-group">
                   <div className="col-sm-offset-2 col-sm-6">
-                      <button type="submit" className="btn btn-success">Search</button>
+                      <button type="submit" className="btn btn-success" onClick={this.searchByProgram}>Search</button>
                       <button type="reset" onClick={this.resetFields} className="btn btn-default cancelBtn">Reset</button>
                   </div>
               </div>
```

Comment:
See my comment above.  I am however planning to raise a ticket that covers the cosmetic changes as the one you have pointed out.  At the moment, only the selected parts are the bits I have worked on.  I will attach the .zip file shortly.

Reply:
My view is that it would be ok for things like comments since they are in no way impacting the functionality. Also if you are working on a file and you could make the code better by just changing a variable name to something more representative, I don't think that would be an issue since just like comments, you are not changing any functionality. These are however just my personal opinions. I think it would help getting @dkayiwa's opinion on this

Reply:
Daniel stated that it is prudent to raise a PR for the direct issue,  and then raise another PR for cosmetic changes, such as the ones you have pointed out above. If you remember, there is a previous ticket that I split into two PRs for the same reason. 
I would recommend that for this PR we focus on just what I changed, then when I make the cosmetic changes, we can address this on that forum @cnagadya 

What are your thoughts on this @dkayiwa ?

Reply:
Code can be tricky in the sense that what appeared to be a mere renaming of a variable may end up introducing bugs. So we prefer having separate tickets and pull requests for any such work. That way, if we end up reverting the pull request because of any subsequent bugs, we do not lose the fix that the pull request was meant for.

Reply:
Noted